pkg/apiserver/registry/inventory/rest.go

```diff
@@ -103,19 +107,29 @@ func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (
 	}
 
 	name := ""
+	namespace := ""
 	if options != nil && options.FieldSelector != nil && options.FieldSelector.String() != "" {
 		fieldSelectorStrings := strings.Split(options.FieldSelector.String(), ",")
 		for _, fieldSelectorString := range fieldSelectorStrings {
 			fieldKeyAndValue := strings.Split(fieldSelectorString, "=")
 			if fieldKeyAndValue[0] == "metadata.name" {
 				name = fieldKeyAndValue[1]
+			} else if fieldKeyAndValue[0] == "metadata.namespace" {
+				namespace = fieldKeyAndValue[1]
 			} else {
-				return nil, errors.NewBadRequest("unsupported field selector, supported labels is: metadata.name")
+				return nil, errors.NewBadRequest("unsupported field selector, supported labels are: metadata.name and metadata.namespace")
 			}
 		}
 	}
 
 	ns, _ := request.NamespaceFrom(ctx)
+	if ns != metav1.NamespaceDefault && namespace != "" && ns != namespace {
+		return nil, errors.NewBadRequest("namespace from field and namespace are different")
```

Comment:
nit: namespace in field selector is different from namespace filter

Reply:
solved